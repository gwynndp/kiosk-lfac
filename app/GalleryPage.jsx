import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import SideBar from './Sidebar';
import MainNav from './MainNav';
import Gallery from './Gallery';

const GalleryWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  min-width: 70%;
`;

const GalleryHeader = styled.header`
  background-color: rgba(255, 255, 255, 0.5);
  margin: 0 0 10px;
  display: flex;
  justify-content: center;
  text-align: center;
  font-size: 24px;
`;

export default function GalleryPage(props) {
  return (
    <React.Fragment>
      <MainNav gallery='true' />
      <GalleryWrapper>
        <GalleryHeader>
          <h1>Meet the People</h1>
        </GalleryHeader>
        <Gallery
          videos={props.videos}
          handleVideoCardClick={props.handleVideoCardClick}
        />
      </GalleryWrapper>
      <SideBar />
    </React.Fragment>
  );
}

GalleryPage.propTypes = {
  videos: PropTypes.arrayOf(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number])
  ).isRequired,
  handleVideoCardClick: PropTypes.func.isRequired,
};
