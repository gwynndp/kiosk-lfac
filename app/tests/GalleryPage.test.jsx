import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import GalleryPage from "../GalleryPage";

test("should render a GalleryPage", () => {
  const props = {
    videos: [
      {
        id: 4,
        category: "people",
        name: "Ashley Sanchez",
        src: "videos/Ashley-Sanchez.mp4",
        img: "images/videos/Ashley-Sanchez.jpg"
      },
      {
        id: 5,
        category: "people",
        name: "Jackie Rose Wilson",
        src: "videos/Jackie-Rose-Wilson.mp4",
        img: "images/videos/Jackie-Rose-Wilson.jpg"
      },
      {
        id: 6,
        category: "people",
        name: "Jude Gibson Byers",
        src: "videos/Jude-Gibson-Byers.mp4",
        img: "images/videos/Jude-Gibson-Byers.jpg"
      },
      {
        id: 7,
        category: "fires",
        name: "Kathryn Hecht",
        src: "videos/Kathryn-Hecht.mp4",
        img: "images/videos/Kathryn-Hecht.jpg"
      },
      {
        id: 8,
        category: "fires",
        name: "Leanne Fogg",
        src: "videos/Leanne-Fogg.mp4",
        img: "images/videos/Leanne-Fogg.jpg"
      }
    ],
    handleVideoCardClick: () => {}
  };

  const tree = renderer
    .create(
      <Router>
        <GalleryPage
          videos={props.videos}
          handleVideoCardClick={props.handleVideoCardClick}
        />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// should render a MainNav
// should render a Gallery
// should render a Sidebar
// should render the correct title

// should direct to a VideoPage when clicked  //DOES THIS TEST GO HERE?
