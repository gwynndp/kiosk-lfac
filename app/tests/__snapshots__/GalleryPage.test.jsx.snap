// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a GalleryPage 1`] = `
Array [
  <div
    className="sc-AykKE heQrPx"
  >
    <a
      className="sc-AykKF igZHWa"
      exact="true"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="home"
        src="images/kiosk_home.png"
        width="100px"
      />
    </a>
    <button
      className="sc-AykKG kmyBlL"
      id="prev"
      onClick={[Function]}
    >
      <div>
        PREV
      </div>
      <img
        alt="scroll up"
        src="images/kiosk_arrow_up.png"
        width="100px"
      />
    </button>
    <button
      className="sc-AykKG kmyBlL"
      id="next"
      onClick={[Function]}
    >
      <img
        alt="scroll down"
        src="images/kiosk_arrow_down.png"
        width="100px"
      />
      <div>
        NEXT
      </div>
    </button>
  </div>,
  <div
    className="sc-fzXfLO ithJqC"
  >
    <header
      className="sc-fzXfLP kFpmZD"
    >
      <h1>
        Meet the People
      </h1>
    </header>
    <div
      className="sc-AykKK ivNex"
      id="gallery"
    >
      <a
        href="/video"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="sc-AykKJ iyPTLm"
          onClick={[Function]}
        >
          <h1
            className="sc-AykKH gGLGOn"
          >
            Ashley Sanchez
          </h1>
          <img
            alt="Video titled Ashley Sanchez"
            className="sc-AykKI kjrCmu"
            src="images/videos/Ashley-Sanchez.jpg"
          />
        </div>
      </a>
      <a
        href="/video"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="sc-AykKJ iyPTLm"
          onClick={[Function]}
        >
          <h1
            className="sc-AykKH gGLGOn"
          >
            Jackie Rose Wilson
          </h1>
          <img
            alt="Video titled Jackie Rose Wilson"
            className="sc-AykKI kjrCmu"
            src="images/videos/Jackie-Rose-Wilson.jpg"
          />
        </div>
      </a>
      <a
        href="/video"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="sc-AykKJ iyPTLm"
          onClick={[Function]}
        >
          <h1
            className="sc-AykKH gGLGOn"
          >
            Jude Gibson Byers
          </h1>
          <img
            alt="Video titled Jude Gibson Byers"
            className="sc-AykKI kjrCmu"
            src="images/videos/Jude-Gibson-Byers.jpg"
          />
        </div>
      </a>
      <a
        href="/video"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="sc-AykKJ iyPTLm"
          onClick={[Function]}
        >
          <h1
            className="sc-AykKH gGLGOn"
          >
            Kathryn Hecht
          </h1>
          <img
            alt="Video titled Kathryn Hecht"
            className="sc-AykKI kjrCmu"
            src="images/videos/Kathryn-Hecht.jpg"
          />
        </div>
      </a>
      <a
        href="/video"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="sc-AykKJ iyPTLm"
          onClick={[Function]}
        >
          <h1
            className="sc-AykKH gGLGOn"
          >
            Leanne Fogg
          </h1>
          <img
            alt="Video titled Leanne Fogg"
            className="sc-AykKI kjrCmu"
            src="images/videos/Leanne-Fogg.jpg"
          />
        </div>
      </a>
      <div
        className="spacer"
        style={
          Object {
            "flexGrow": "1",
          }
        }
      />
    </div>
  </div>,
  <div
    className="sc-AykKC gVzywQ"
  >
    <div
      id="logos-wrapper"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": "0 10px 20px 0",
        }
      }
    >
      <img
        alt="Cloverdale City logo"
        src="images/cloverdale-city-logo.png"
        width="100%"
      />
      <img
        alt="Cloverdale High School logo"
        src="images/cloverdale-hs-logo.png"
        width="100%"
      />
      <img
        alt="Listening For a Change logo"
        src="images/lfac_logo_black.gif"
        width="100%"
      />
    </div>
    <a
      className="sc-AykKD bhXdAw"
      href="/lfac"
      onClick={[Function]}
    >
      <span>
        Listening for a Change
      </span>
    </a>
    <a
      className="sc-AykKD bhXdAw"
      href="/schools"
      onClick={[Function]}
    >
      <span>
        Schools
      </span>
    </a>
    <a
      className="sc-AykKD bhXdAw"
      href="/fires"
      onClick={[Function]}
    >
      <span>
        Fire Stories
      </span>
    </a>
    <a
      className="sc-AykKD bhXdAw"
      href="/jackdemayo"
      onClick={[Function]}
    >
      <span>
        Remembering Jack DeMayo
      </span>
    </a>
    <a
      className="sc-AykKD bhXdAw"
      href="/vom"
      onClick={[Function]}
    >
      <span>
        Valley of the Moon
      </span>
    </a>
  </div>,
]
`;
